python/python/lance/dataset.py

```diff
@@ -2298,8 +2299,34 @@ def optimize_indices(self, **kwargs):
         the new data to existing partitions.  This means an update is much quicker
         than retraining the entire index but may have less accuracy (especially
         if the new data exhibits new patterns, concepts, or trends)
-        """
```

Comment:
Can we update some of this wording so that we also explain what "merging" is and why you would want to do it?